### PR TITLE
Internal links on site head

### PR DESCRIPTION
**Pull request by spi43984**
I needed some links on the cover page to link to internal sites and not showing the leading fork awesome icon.
For some of these pages I also wanted to get the content still rendered on the homepage - for others not.

With these changes two parameters are introduced in the markdown files:
internal = the URL to link to
internal_hide = if true then do not render this markdown file (as a side-effect this works on all homepage markdown files to show or hide them)

### CODE REVIEW

File: exampleSite/content/homepage/internal.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+title: "Link2InternalPage"
```
Comment (zjedi) [inline exampleSite/content/homepage/internal.md:1]: What's the use case? Linking to single, dedicated page?

Comment (spi43984) [reply exampleSite/content/homepage/internal.md:1]: I needed an internal, dedicated page for detailed contact and legal information. An external link I didn't like due to the leading icon. BTW - I had to set up a 1st markdown file with an external link, then I needed the single page with the legal staff (2nd markdown file). And I wanted to have the legal stuff shortened as a section on my homepage (the 3rd markdown file), but without an additional button on the cover page (as I already had one for the dedicated page, but on the other hand I wanted a navigation menu entry for the section if the user scrolled down a little).

As I didn't like the icon for the external link button I created code for the internal link. While creating this code I figured out that I just need two markdown files - the single page and just another one with the internal link destination and the content for the section. That's why I introduced the parameter internal_hide - now I can add multiple links as buttons on the cover page and have a section on the homepage for some and not for the others.

**Review by zjedi — CHANGES_REQUESTED** [review]
Something seems wrong with the demo page, see deployed demo, at the very bottom. https://deploy-preview-158--hugo-scroll.netlify.app/#github
image

**Comment by spi43984** [comment]
> Something seems wrong with the demo page, see deployed demo, at the very bottom. https://deploy-preview-158--hugo-scroll.netlify.app/#github image

Added leading '---' to internal.md.

**Review by zjedi — APPROVED** [review]

**Comment by zjedi** [comment]
@spi43984 If you go check the demo now, there are some outstanding issues
- two sections Contact + Contacts, sounds illogical to have two of them
- try navigating from landing page, from left "sticky menu", play around, I managed to run into an error.

**Comment by spi43984** [comment]
> @spi43984 If you go check the demo now, there are some outstanding issues
> 
> * two sections Contact + Contacts, sounds illogical to have two of them

fair point - as there is already a contact section, will prepare another example, maybe legal or something

> * try navigating from landing page, from left "sticky menu", play around, I managed to run into an error.

What was the error? Can you please give some more context?
Or is it GITHUB to show up in the nav menu? Which should not as it is an external link, found a bug in index.html - am trying to solve that:
```
{{ $content := where $sections "Params.external" "==" nil }}
{{ $content := where $sections "Params.internal_hide" "ne" true }}
```
The second line circumvents the first one and renders homepage content even it is an external link.

Is there a combination possible like
```
{{ $content := where $sections and ("Params.internal_hide" "ne" true) ("Params.internal_hide" "ne" true) }}
```

**Comment by spi43984** [comment]
> @spi43984 If you go check the demo now, there are some outstanding issues
> 
> * two sections Contact + Contacts, sounds illogical to have two of them
> * try navigating from landing page, from left "sticky menu", play around, I managed to run into an error.

please review

**Comment by zjedi** [comment]
> two sections Contact + Contacts, sounds illogical to have two of them

Looks good now, thanks!

> try navigating from landing page, from left "sticky menu", play around, I managed to run into an error.

Unable to reproduce the error, looks ok now.

File: exampleSite/content/homepage/internal.md
Comment (zjedi) [inline exampleSite/content/homepage/internal.md:1]: could this file be renamed to something like `legal-brief.md` to make it more obvious?

File: exampleSite/content/homepage/internal.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+title: "Brief Legal Information"
+weight: 98
+header_menu_title: "Legal"
+navigation_menu_title: "Legal stuff"
+header_menu: true
+internal: /legal/
```
Comment (zjedi) [inline exampleSite/content/homepage/internal.md:7]: Something like `detailed_page_path: /legal/` would sound more descriptive :)

then also:
`internal_hide: false` -> or `detailed_page_exclude_homepage_content: false` or maybe even better  `detailed_page_homepage_content: true` (simplify double negation)

**Review by zjedi — CHANGES_REQUESTED** [review]
Looks good, just a couple suggestions regarding configuration property names to make them more obvious :)

File: exampleSite/content/homepage/internal.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+internal_hide: false
+---
+The file **internal.md** creates an internal link on the cover page without the leading icon.
+
+If the parameter `internal_hide` is set to true neither this section nor the navigation menu entry will be shown (but the link on the cover page will be).
+
+If `internal_hide` is set to false, the link, navigation menu and the content will be rendered. In that case thids section might contain less or just summarized information compared to the single page referenced by the parameter `internal`.
```
Comment (zjedi) [inline exampleSite/content/homepage/internal.md:14]: typo: thids ;)

File: exampleSite/content/homepage/internal.md
Comment (spi43984) [reply exampleSite/content/homepage/internal.md:1]: ok

File: exampleSite/content/homepage/internal.md (hunk 3 above)
Comment (spi43984) [reply exampleSite/content/homepage/internal.md:14]: ok

**Review by zjedi — APPROVED** [review]

**Pull request merged**